File: litho-rendercore/src/test/java/com/facebook/rendercore/SimpleTestBinder.java
// (c) Meta Platforms, Inc. and affiliates. Confidential and proprietary.

package com.facebook.rendercore;

import android.content.Context;
import androidx.annotation.Nullable;

final class SimpleTestBinder implements RenderUnit.Binder<Object, Object, Void> {
  final Runnable mMountRunnable;

  SimpleTestBinder(Runnable mountRunnable) {
    mMountRunnable = mountRunnable;
  }

  @Override
  public boolean shouldUpdate(
      Object currentModel,
      Object newModel,
      @Nullable Object currentLayoutData,
      @Nullable Object nextLayoutData) {
    return true;
  }

  @Override
  public Void bind(Context context, Object o, Object o2, @Nullable Object layoutData) {
    mMountRunnable.run();
    return null;
  }

  @Override
  public void unbind(
      Context context, Object o, Object o2, @Nullable Object layoutData, Void bindData) {}
}

File: litho-rendercore/src/test/java/com/facebook/rendercore/SimpleTestBinder.kt
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.facebook.rendercore

import android.content.Context

internal class SimpleTestBinder(private val mountRunnable: Runnable) :
    RenderUnit.Binder<Any?, Any, Any> {

  override fun shouldUpdate(
      currentModel: Any?,
      newModel: Any?,
      currentLayoutData: Any?,
      nextLayoutData: Any?
  ): Boolean = true

  override fun bind(context: Context, content: Any, model: Any?, layoutData: Any?): Any? {
    mountRunnable.run()
    return null
  }

  override fun unbind(
      context: Context,
      content: Any,
      model: Any?,
      layoutData: Any?,
      bindData: Any?
  ) = Unit

  override val description: String
    get() = super.description
}
